File: 351004/Nepesov/Lab2/untitled/src/main/java/org/example/repository/StickerRepository.java
package org.example.repository;

import org.example.model.Sticker;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface StickerRepository extends JpaRepository<Sticker, Long> {
    Optional<Sticker> findByName(String name);
}

File: 351002/Abrashin/src/main/kotlin/com/example/distcomp/repository/StickerRepository.kt
package com.example.distcomp.repository

import com.example.distcomp.model.Sticker
import org.springframework.stereotype.Repository

@Repository
class StickerRepository : InMemoryRepository<Sticker>() {
    fun findByName(name: String) = storage.values.find { it.name == name }
}
